### ISSUE TITLE
Callback didReadData can return data from old connection after reconnect

### ISSUE THREAD

**Issue by misos1** [robbiehanson/CocoaAsyncSocket#576]
When sending data between app and server and then interrupting connection by locking iphone screen (after about 3-4 seconds will ios drop connections) and returning back and reconnect again using connectToHost then didReadData callback can return data from old connection after is established new connection. This can mess up things when lets say parsing data in two stages alternating readDataToLength for reading header of constant size which contain length of rest data and then rest of that data. I did not go deep how can this happen and it is rather rare. Quick workaround is to always creating new socket for new connection after we were disconnected and use socket which is passed as parameter into didReadData or check this parameter against our current socket and did not do anything if data goes from old socket.

In other words GCDAsyncSocket can return data to didReadData callback after it reported that socket is disconnected either by isDisconnected true or by socketDidDisconnect (I am not sure exactly which in my case) and even worse can return these old data from old connection after call to connectToHost and even after SSL handshake and another readDataToLength enqueues right after connect.
